cpp/src/db/db_object.cu

```diff
@@ -120,84 +120,62 @@ bool db_pattern<idx_t>::isAllConstants() {
 template class db_pattern<int32_t>;
 template class db_pattern<int64_t>;
 
-template<typename idx_t>
-void db_column_index<idx_t>::deleteData() {
-  if (offsets != nullptr) {
-    ALLOC_FREE_TRY(offsets, nullptr);
-    offsets = nullptr;
-    offsets_size = 0;
-  }
-  if (indirection != nullptr) {
-    ALLOC_FREE_TRY(indirection, nullptr);
-    indirection = nullptr;
-    indirection_size = 0;
-  }
-}
-
 template<typename idx_t>
 db_column_index<idx_t>::db_column_index() {
-  offsets = nullptr;
   offsets_size = 0;
-  indirection = nullptr;
   indirection_size = 0;
 }
 
 template<typename idx_t>
```

Comment:
Most of the codebase is decoupling `idx` type from `offset` type as the latter is expected to get bigger than 2B sooner. Would that be relevant for DB?

Reply:
DB isn't set up to distinguish between offset type and index type, and I don't really think that it needs to be.